tools/dataset/preprocess_h36m.py

```diff
@@ -0,0 +1,336 @@
+        f = metadata_slice[6:8, np.newaxis]
+
+        # distortion
+        k = metadata_slice[10:13, np.newaxis]
+        p = metadata_slice[13:15, np.newaxis]
+
+        return {
+            'R': R,
+            'T': T,
+            'c': c,
+            'f': f,
+            'k': k,
+            'p': p,
+            'name': f'camera{camera + 1}',
+            'id': self.camera_ids[camera]
+        }
+
+    def _load_annotations(self, subject, action, subaction, camera):
+        """Load annotations for a sequence."""
+        subj_dir = join(self.extracted_dir, subject)
+        basename = self._get_base_filename(subject, action, subaction, camera)
+
+        # load 2D keypoints
+        with pycdf.CDF(
+                join(subj_dir, 'MyPoseFeatures', 'D2_Positions',
+                     basename + '.cdf')) as cdf:
+            kps_2d = np.array(cdf['Pose'])
+
+        num_frames = kps_2d.shape[1]
+        kps_2d = kps_2d.reshape((num_frames, 32, 2))[::self.sample_rate,
+                                                     self.movable_joints]
+        kps_2d = np.concatenate([kps_2d, np.ones((len(kps_2d), 17, 1))],
+                                axis=2)
+
+        # load 3D keypoints
+        with pycdf.CDF(
+                join(subj_dir, 'MyPoseFeatures', 'D3_Positions_mono',
+                     basename + '.cdf')) as cdf:
+            kps_3d = np.array(cdf['Pose'])
+
+        kps_3d = kps_3d.reshape(
+            (num_frames, 32, 3))[::self.sample_rate,
+                                 self.movable_joints] / 1000.
+        kps_3d = np.concatenate([kps_3d, np.ones((len(kps_3d), 17, 1))],
+                                axis=2)
+
+        # calculate bounding boxes
+        bboxes = np.stack([
+            np.min(kps_2d[:, :, 0], axis=1),
+            np.min(kps_2d[:, :, 1], axis=1),
+            np.max(kps_2d[:, :, 0], axis=1),
+            np.max(kps_2d[:, :, 1], axis=1)
+        ],
+                          axis=1)
+        centers = np.stack([(bboxes[:, 0] + bboxes[:, 2]) / 2,
+                            (bboxes[:, 1] + bboxes[:, 3]) / 2],
+                           axis=1)
+        scales = self.scale_factor * np.max(
+            bboxes[:, 2:] - bboxes[:, :2], axis=1) / 200
+
+        # extract frames and save imgnames
+        imgnames = []
+        video_path = join(subj_dir, 'Videos', basename + '.mp4')
+        sub_base = subject + '_' + basename.replace(' ', '_')
+        img_dir = join(self.processed_dir, 'imgs', subject, sub_base)
+        os.makedirs(img_dir, exist_ok=True)
+        prefix = join(subject, sub_base, sub_base)
+
+        cap = cv2.VideoCapture(video_path)
+        for i in range(0, num_frames, self.sample_rate):
+            imgname = prefix + f'_{i + 1: 06d}.jpg'
+            imgnames.append(imgname)
+            dest_path = join(self.processed_dir, 'imgs', imgname)
+            if os.path.exists(dest_path):
+                continue
+            cap.set(1, i)
```

Comment:
Generally speaking, `.set(CAP_PROP_POS_FRAMES, frameNumber)` is not guaratted to be exact. Sequentially decoding all frames and keep the desired ones is more safe.

Btw, does here need exact frame seeking?

Reply:
The FPS is 50, so neighboring frames may have little difference. But I think exact frame seeking is better.